src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2159,6 +2159,7 @@ class SelectStatementSegment(BaseSegment):
     NonWithNonSelectableGrammar=OneOf(
         Ref("UpdateStatementSegment"),
         Ref("InsertStatementSegment"),
+        Ref("DeleteStatementSegment"),
```

Comment:
This is what allows DELETE in WITH statements (see #1889 for originial UPDATE/INSERT PR)